src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -114,6 +115,74 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
                 // Start computing the initial set of tags immediately.  We want to get the UI
                 // to a complete state as soon as possible.
                 ComputeInitialTags();
+
+                return;
+
+                void Connect()
+                {
+                    this.AssertIsForeground();
+
+                    _eventSource.Changed += OnEventSourceChanged;
+
+                    if (_dataSource.TextChangeBehavior.HasFlag(TaggerTextChangeBehavior.TrackTextChanges))
+                    {
+                        _subjectBuffer.Changed += OnSubjectBufferChanged;
+                    }
+
+                    if (_dataSource.CaretChangeBehavior.HasFlag(TaggerCaretChangeBehavior.RemoveAllTagsOnCaretMoveOutsideOfTag))
+                    {
+                        if (_textViewOpt == null)
+                        {
+                            throw new ArgumentException(
+                                nameof(_dataSource.CaretChangeBehavior) + " can only be specified for an " + nameof(IViewTaggerProvider));
+                        }
+
+                        _textViewOpt.Caret.PositionChanged += OnCaretPositionChanged;
+                    }
+
+                    // Tell the interaction object to start issuing events.
+                    _eventSource.Connect();
+                }
+            }
```

Comment:
'Connect' moved into the constructor (the only place it is allowed to be called).
'Dispose' moved next to hte constructor to make the symmetries between construction and disposal apparant.
'Diconnect' moved into 'Dispose' (the only place it is allowed to be called).

I kept Connect/Disconnect (even though they could be inlined) as they have a nice symmetry and one clearly reverses the other.